README.md

```diff
@@ -169,10 +200,32 @@ so you need to define all of them:
    Either keep a count of errors (abort for >2 failed attempts) or use a timeout for `ap_room_info` (abort after >5sec).
 * `ap_bounced(bounce: json)` called when receiving Bounced from server. Only for `api_version >= 2`.
 
+These events also make the following JSON data available through the `apclient_json_...` calls:
+
+* `ap_slot_refused` { "len": `int`, "reasons" : `str[]` }
+* `ap_slot_connected` "slot_data" of [Connected](https://github.com/ArchipelagoMW/Archipelago/blob/main/docs/network%20protocol.md#connected)
+* `ap_items_received` { "index": `int`, "len": `int`, "ids": `int[]`, "names": `str[]`, "flags": `int[]`, "players": `int[]`, "locations": `int[]` }
+* `ap_location_info` { "len": `int`, "items": `int[]`, "flags": `int[]`, "players": `int[]`, "locations": `int[]` }
+* `ap_location_checked` { "len": `int`, "locations": `int[]` }
+* `ap_print_json` see [PrintJSON](https://github.com/ArchipelagoMW/Archipelago/blob/main/docs/network%20protocol.md#printjson)
+* `ap_socket_error` { "message": `str` }
+* `ap_bounced` see [Bounced](https://github.com/ArchipelagoMW/Archipelago/blob/main/docs/network%20protocol.md#bounced)
+
+Additionally, these built-in GM functions may be called by an `apclient_poll` script:
+
+* `show_message(message: str): none` when an exception occurs.
+
+If this happens, the following JSON data is made available:
+
+* `show_message` { "message": `str` }
+
 ### Not implemented
 
-The following are not implemented for `api_version = 1`.
+The following are not implemented.
 
-* `ap_bounced(...)` when a player sent a Bounce
 * `ap_retrieved(...)` result for `apclient_get`
 * `ap_set_reply(...)` when a player sent a Set that matches your SetNotify
+
+The following are only implemented for `api_version >= 2`.
+
+* `ap_bounced(...)` when a player sent a Bounce
```

Comment:
Looking at [the final render](https://github.com/black-sliver/gm-apclientpp/blob/4e35538284cd459a585c49640dd453ccfa1b20bc/README.md), I realized that this section is now under the heading "Not implemented". So this should probably move up by a couple of lines above to `### Not implemented`